### PR TITLE
Fix number of input channels for controlnet conditioning

### PR DESCRIPTION
**Pull request by jfozard**
The FlaxControlNetConditioningEmbedding has the number of input channels fixed at 3. This is hard-coded in the init_weights method https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/controlnet_flax.py#L173

We need to supply 6 channels. Added code to change the dimensions of the input conv kernel parameter after initialisation.

Also fixed stacking of input tensors.

### CODE REVIEW

**Comment by jfischoff** [comment]
Here are the changes I had to make controlnet.py file to thread the conditioning channel count to unet:
```patch
+++ b/src/diffusers/models/controlnet.py
@@ -119,6 +119,7 @@ class ControlNetModel(ModelMixin, ConfigMixin):
         projection_class_embeddings_input_dim: Optional[int] = None,
         controlnet_conditioning_channel_order: str = "rgb",
         conditioning_embedding_out_channels: Optional[Tuple[int]] = (16, 32, 96, 256),
+        conditioning_channels: int = 3,
     ):
         super().__init__()
 
@@ -184,6 +185,7 @@ class ControlNetModel(ModelMixin, ConfigMixin):
         self.controlnet_cond_embedding = ControlNetConditioningEmbedding(
             conditioning_embedding_channels=block_out_channels[0],
             block_out_channels=conditioning_embedding_out_channels,
+            conditioning_channels=conditioning_channels
         )
 
         self.down_blocks = nn.ModuleList([])
@@ -265,6 +267,7 @@ class ControlNetModel(ModelMixin, ConfigMixin):
         controlnet_conditioning_channel_order: str = "rgb",
         conditioning_embedding_out_channels: Optional[Tuple[int]] = (16, 32, 96, 256),
         load_weights_from_unet: bool = True,
+        conditioning_channels: int = 3,
     ):
         r"""
         Instantiate Controlnet class from UNet2DConditionModel.
@@ -297,6 +300,7 @@ class ControlNetModel(ModelMixin, ConfigMixin):
             projection_class_embeddings_input_dim=unet.config.projection_class_embeddings_input_dim,
             controlnet_conditioning_channel_order=controlnet_conditioning_channel_order,
             conditioning_embedding_out_channels=conditioning_embedding_out_channels,
+            conditioning_channels=conditioning_channels,
         )
```

File: src/train_controlnet_flax.py [hunk 1]
```diff
@@ -806,7 +807,9 @@ def main():
             "mid_block",
         ]:
             controlnet_params[key] = unet_params[key]
-
+        rng, rng_params = jax.random.split(rng)
+        controlnet_params['controlnet_cond_embedding']['conv_in']['kernel']=jax.nn.initializers.lecun_normal()(rng_params, (3,3,6,16), jnp.float32)
```
Comment (jfischoff) [inline src/train_controlnet_flax.py:811]: The changes I made also affected the `controlnet_cond_embedding` but during initialization. I'm not sure if this identical. I'll try to confirm, but can you as well?

Comment (jfischoff) [reply src/train_controlnet_flax.py:811]: I think doing it this way is error prone. Looking at the code of path of initialization I can see that in channels are used to initialize several things. Also, I haven't found a property called kernel yet. 

What are your thoughts?

File: src/train_controlnet_flax.py [hunk 2]
```diff
@@ -622,15 +622,16 @@ def preprocess_train(examples):
         images = [image_transforms(image) for image in images]
 
         conditioning_images_0 = [image.convert("RGB") for image in examples[conditioning_image_column]]
-        conditioning_images_0 = [conditioning_image_transforms(image) for image in conditioning_images_0]
+        conditioning_images_0 = torch.stack([conditioning_image_transforms(image) for image in conditioning_images_0], dim=0)
```
Comment (jfischoff) [inline src/train_controlnet_flax.py:625]: what is the torch.stack for?

Comment (jfozard) [reply src/train_controlnet_flax.py:625]: torch.cat in  conditioning_images = torch.cat([conditioning_images_0, conditioning_images_1], dim=1) requires Tensors as conditioning_images_0 and _1, not lists

File: src/train_controlnet_flax.py (hunk 1 above)
Comment (jfozard) [reply src/train_controlnet_flax.py:811]: The JaX/Flax model doesn't use src/diffusers/models/controlnet.py - it's all rewritten in JaX/Flax in src/diffusers/models/controlnet_flax.py
This new solution is suboptimal, but works without making changes to the underlying library.

Comment (jfischoff) [reply src/train_controlnet_flax.py:811]: Okay, I'll try to double check the jax code. If we can avoid changing the underlying library, great. If we have to change it, we have to change it
